pocs/tests/test_filterwheel.py

```diff
@@ -0,0 +1,148 @@
+from pocs.camera.simulator import Camera as SimCamera
+
+
+@pytest.fixture(scope='module')
+def filterwheel():
+    sim_filterwheel = SimFilterWheel(filter_names=['one', 'deux', 'drei', 'quattro'],
+                                     move_time=0.1 * u.second,
+                                     timeout=0.5 * u.second)
+    return sim_filterwheel
+
+# intialisation
+
+
+def test_init(filterwheel):
+    assert isinstance(filterwheel, SimFilterWheel)
+    assert filterwheel.is_connected
+
+
+def test_camera_init():
+    sim_camera = SimCamera(filterwheel={'model': 'simulator',
+                                        'filter_names': ['one', 'deux', 'drei', 'quattro']})
+    assert isinstance(sim_camera.filterwheel, SimFilterWheel)
+    assert sim_camera.filterwheel.is_connected
+    assert sim_camera.filterwheel.uid
+    assert sim_camera.filterwheel.camera is sim_camera
+
+
+def test_camera_association_on_init():
+    sim_camera = SimCamera()
+    sim_filterwheel = SimFilterWheel(filter_names=['one', 'deux', 'drei', 'quattro'],
+                                     camera=sim_camera)
+    assert sim_filterwheel.camera is sim_camera
+
+
+def test_with_no_name():
+    with pytest.raises(ValueError):
+        sim_filterwheel = SimFilterWheel()
+
+# Basic property getting and (not) setting
+
+
+def test_model(filterwheel):
+    model = filterwheel.model
+    assert model == 'simulator'
+    with pytest.raises(AttributeError):
+        filterwheel.model = "Airfix"
+
+
+def test_name(filterwheel):
+    name = filterwheel.name
+    assert name == 'Simulated Filter Wheel'
+    with pytest.raises(AttributeError):
+        filterwheel.name = "Phillip"
+
+
+def test_uid(filterwheel):
+    uid = filterwheel.uid
+    assert uid.startswith('SW')
+    assert len(uid) == 6
+    with pytest.raises(AttributeError):
+        filterwheel.uid = "Can't touch this"
+
+
+def test_filter_names(filterwheel):
+    names = filterwheel.filter_names
+    assert isinstance(names, list)
+    for name in names:
+        assert isinstance(name, str)
+    with pytest.raises(AttributeError):
+        filterwheel.filter_names = ["Unsharp mask", "Gaussian blur"]
+
+# Movement
+
+
+def test_move_number(filterwheel):
+    assert filterwheel.position == 1
+    e = filterwheel.move_to(2)
+    assert math.isnan(filterwheel.position)  # position is NaN while between filters
+    e.wait()
+    assert filterwheel.position == 2
+    e = filterwheel.move_to(3, blocking=True)
+    assert e.is_set()
+    assert filterwheel.position == 3
+    filterwheel.position = 4  # Move by assignment to position property blocks until complete
+    assert filterwheel.position == 4
+
+
+def test_move_bad_number(filterwheel):
+    with pytest.raises(ValueError):
+        filterwheel.move_to(0, blocking=True)  # No zero based numbering here!
+    with pytest.raises(ValueError):
+        filterwheel.position = 99
+
+
+def test_move_name(filterwheel, caplog):
```

Comment:
Wha?! How did I not know about `caplog` fixture?! That's nice.

Reply:
It is. I was forced into finding out how to do this because exceptions raised in a `Thread` don't make it out to the calling thread, so the log messages were the only way I could test that the timeout was happening in `test_move_timeout`.